src/hotspot/os/bsd/os_bsd.cpp

```diff
@@ -2142,10 +2183,15 @@ static int anon_munmap(char * addr, size_t size) {
   return ::munmap(addr, size) == 0;
 }
 
-char* os::pd_reserve_memory(size_t bytes, char* requested_addr,
+MACOS_ONLY(char* os::pd_reserve_memory(size_t bytes, char* requested_addr,
+                            size_t alignment_hint,
+                            bool executable) {
+  return anon_mmap(requested_addr, bytes, (requested_addr != NULL), executable);
+})
+NOT_MACOS(char* os::pd_reserve_memory(size_t bytes, char* requested_addr,
                             size_t alignment_hint) {
   return anon_mmap(requested_addr, bytes, (requested_addr != NULL));
-}
+})
```

Comment:
Nitpicking and a matter of taste, but for multiline diffs I find #ifdef __APPLE__ easier to read.